packages/c/sshnpd/src/run_srv_process.c

```diff
@@ -11,22 +10,22 @@
 
 #define LOGGER_TAG "RUN SRV"
 
-int run_srv_process(sshnpd_params *params, cJSON *host, cJSON *port, bool npt, cJSON *requested_host,
-                    cJSON *requested_port, bool authenticate_to_rvd, char *rvd_auth_string, bool encrypt_rvd_traffic,
-                    bool multi, unsigned char *session_aes_key_encrypted, unsigned char *session_iv_encrypted) {
+int run_srv_process(const char *srvd_host, uint16_t srvd_port, const char *requested_host, uint16_t requested_port,
+                    bool authenticate_to_rvd, char *rvd_auth_string, bool encrypt_rvd_traffic, bool multi,
+                    unsigned char *session_aes_key_encrypted, unsigned char *session_iv_encrypted) {
 
   int res = 0;
   srv_params_t srv_params;
   apply_default_values_to_srv_params(&srv_params);
 
-  srv_params.host = cJSON_GetStringValue(host);
-  srv_params.port = (uint16_t)cJSON_GetNumberValue(port);
+  srv_params.host = (char *)srvd_host;
+  srv_params.port = srvd_port;
 
-  if (npt) {
-    srv_params.local_host = cJSON_GetStringValue(requested_host);
-    srv_params.local_port = (uint16_t)cJSON_GetNumberValue(requested_port);
-  } else {
-    srv_params.local_port = params->local_sshd_port;
+  if (requested_host != NULL) {
+    srv_params.local_host = (char *)requested_host;
+  }
+  if (requested_port != 0) {
+    srv_params.local_port = requested_port;
   }
```

Comment:
replaced sshnpd dependant logic, with simpler behavior, srv_params.local_host and srv_params.local_port already have default values of `localhost` and `22` in srv

Reply:
nice